class.jetpack.php

```diff
@@ -3018,7 +2891,7 @@ function admin_init() {
 		// If the plugin is not connected, display a connect message.
 		if (
 			// the plugin was auto-activated and needs its candy
-			Jetpack_Options::get_option( 'do_activate' )
+			Jetpack_Options::get_option_and_ensure_autoload( 'do_activate', '0' )
```

Comment:
Unfortunately the functionality is a little bit incompatible. For example, there's no sensible way to ensure autoload for grouped options. Also some of the options don't start with `jetpack_` (e.g. `jetpack-twitter-cards-site-tag`, `open_graph_protocol_site_type`), and we would have had to change the values of some constants that might be used outside Jetpack. That's why I made it separate.

It's not the cleanest solution, I know :/ I'm open to suggestions...